api/src/main/java/org/confeti/controllers/core/ReportController.java

```diff
@@ -37,46 +40,87 @@
+            @RequestParam(value = YEAR_URI_PARAMETER, required = false) final Optional<Integer> year) {
+        return handleRequest(year, this::languageToSetConverter);
+    }
+
+    private Mono<ResponseEntity<?>> handleRequest(final int year,
+                                                  final String conferenceName,
+                                                  final Function<Report, Collection<String>> reportConverter) {
+        return countInfo(reportService.findBy(conferenceName, year), reportConverter)
+                .<ResponseEntity<?>>map(map -> ResponseEntity.ok(new ReportResponse(conferenceName, Map.of(year, map))))
+                .onErrorResume(Exception.class, err -> Mono.just(ResponseEntity.badRequest().body(new ErrorResponse(err.getMessage()))));
+    }
+
+    private Mono<ResponseEntity<?>> handleRequest(final String conferenceName,
+                                                  final Function<Report, Collection<String>> reportConverter) {
+        return countInfoByYear(reportService.findBy(conferenceName)
+                        .groupBy(report -> report.getConferences().iterator().next().getYear()), /* ATTENTION: in this case Set consists only 1 element */
+                reportConverter)
+                .map(map -> new ReportResponse(conferenceName, map))
+                .<ResponseEntity<?>>map(ResponseEntity::ok)
+                .onErrorResume(Exception.class, err -> Mono.just(ResponseEntity.badRequest().body(new ErrorResponse(err.getMessage()))));
+    }
+
+    private Mono<ResponseEntity<?>> handleRequest(final Optional<Integer> year, final Function<Report, Collection<String>> reportConverter) {
         return reportService.findAll()
                 .flatMap(report -> Flux.fromIterable(report.getConferences())
                         .filter(conference -> year.isEmpty() || year.get().equals(conference.getYear()))
                         .zipWith(Mono.just(report)))
                 .groupBy(t -> t.getT1().getName())
-                .flatMap(group -> countTagsByYear(group.groupBy(t -> t.getT1().getYear(), Tuple2::getT2))
-                        .map(map -> new TagResponse(group.key(), map)))
+                .flatMap(group -> countInfoByYear(group.groupBy(t -> t.getT1().getYear(), Tuple2::getT2), reportConverter)
+                        .map(map -> new ReportResponse(group.key(), map)))
                 .collectList()
                 .<ResponseEntity<?>>map(ResponseEntity::ok)
                 .onErrorResume(Exception.class, err -> Mono.just(ResponseEntity.badRequest().body(new ErrorResponse(err.getMessage()))));
     }
 
-    private static Mono<Map<String, Long>> countTags(final Flux<Report> reports) {
-        return reports.flatMapIterable(report -> Objects.requireNonNullElse(report.getTags(), Collections.emptySet()))
+    private static Mono<Map<String, Long>> countInfo(final Flux<Report> reports, Function<Report, Collection<String>> converter) {
```

Comment:
I think that the `converter` should be the `reportConverter` as in other methods.